### DISCUSSION TITLE
The general format of JSON workflow is a horrendous thing to work with as a developer

### DISCUSSION THREAD

**Discussion by BrechtCorbeel** [Comfy-Org/ComfyUI#4787] [General]
Now maybe I will get a more thorough understanding of how it is structured that makes it easier to manipulate, but:

- the workflow contains multiple times the same values for the front end and the backend and the widget_values and the values specified can thus be different
- I can imagine this being a security loophole as someone might create workflows that look ok on the front, but are malware in the backend
- it makes it really hard to change using programs or engineering programs to alter them as there is no true key: value structure to refer to as "widgets_values" can refer to any widget (almost) and is really hard and engineering intensive to get a hold of. Instead of being logical, concise and consistent.

**Comment by ltdrdata** [comment]
I also think a key-value structure would be more desirable.
Due to the current structure, compatibility issues with workflows frequently occur when node shape is updated.
